devito/propagator.py

```diff
@@ -342,11 +342,16 @@ def sympy_to_cgen(self, stencils):
             stencil = self.convert_equality_to_cgen(equality)
             stmts.append(stencil)
 
+        nest_decl = lambda dec, stmt: cgen.Assign(dec.__str__()[:-1], stmt.rvalue)
+
+        for idx, dec in enumerate(decl):
+            stmts[idx] = nest_decl(dec, stmts[idx])
```

Comment:
don't need a lambda function here, as you use it only in one place

you can also do ``str(dec)`` instead of ``dec.__str__``. Why do you strip out the last element (with [:-1] )? What format is that ?